### PR TITLE
Fix azure key vault cleanup failure

### LINKED ISSUE #8493: Key Vaule is not deleted on cleanup azure-arm
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

When using the azure-arm provisioner with build_resource_group_name, the key vault resource isn't deleted at cleanup

#### Reproduction Steps

See overview

### Packer version

1.4.5 and 1.4.4

### Operating system and Environment details

OS, Architecture, and any other information you can provide about the
environment.

Windows Server 2019, 64x, Nothing has been installed on these machines bar packer

### Log Fragments and crash.log files

I was unable to set env variables as the machines are locked down. But the console output lists the keyvault as one of the resources to delete but doesnt do anything. No delete request within azure, nothing blocking the subscription or resource group from deleting resources etc.

### LINKED ISSUE #8903: Azure tech debt: Switch out custom vault client implementation for azure sdk implementation
We've got a TODO here: https://github.com/hashicorp/packer/blob/master/builder/azure/common/vault.go#L1 from 2016; the sdk appears to have added a vault client that same year but we never got around to implementing it. The current vault client doesn't seem to be deleting things correctly, so it's probably time to upgrade.

### PR DESCRIPTION
**Pull request by SwampDragons**
Replace the nonfunctional delete mechanisms in the home-grown keyvault client with the official sdk vault client.

Fix the logic around deciding whether to delete the vault. 

Closes #8493 
Closes #8903

### CODE REVIEW

File: builder/azure/arm/step_delete_resource_group.go [hunk 1]
```diff
@@ -45,7 +45,8 @@ func (s *StepDeleteResourceGroup) deleteResourceGroup(ctx context.Context, state
 
 		if keyVaultDeploymentName, ok := state.GetOk(constants.ArmKeyVaultDeploymentName); ok {
 			// Only delete if custom keyvault was not provided.
-			if exists := state.Get(constants.ArmIsExistingKeyVault).(bool); exists {
+			if exists := state.Get(constants.ArmIsExistingKeyVault).(bool); !exists {
```
Comment (nywilken) [inline builder/azure/arm/step_delete_resource_group.go]: Ah, nice catch here.

File: builder/azure/arm/step_deploy_template.go [hunk 2]
```diff
@@ -117,8 +117,7 @@ func deleteResource(ctx context.Context, client *AzureClient, resourceType strin
 		}
 		return err
 	case "Microsoft.KeyVault/vaults":
-		// TODO(paulmey): not sure why VaultClient doesn't do cancellation
-		_, err := client.VaultClientDelete.Delete(resourceGroupName, resourceName)
+		_, err := client.VaultClientDelete.Delete(ctx, resourceGroupName, resourceName)
```
Comment (nywilken) [inline builder/azure/arm/step_deploy_template.go]: I'll take a context for $2000 :+1:

File: builder/azure/common/vault.go [hunk 3]
```diff
@@ -137,72 +133,6 @@ func (client *VaultClient) SetSecret(vaultName, secretName string, secretValue s
 	return nil
 }
 
-// Delete deletes the specified Azure key vault.
```
Comment (nywilken) [inline builder/azure/common/vault.go]: :open_mouth:

**Comment by dipesharora** [comment]
When can we expect this fix to be released, will it be released with ver 1.5.5 and what is the expected release date?

**Comment by SwampDragons** [comment]
It will be released with v1.5.5. We are hoping to release this week.

**Pull request merged**